SMPL++/src/smpl/SMPL.cpp

```diff
@@ -522,8 +522,25 @@ void SMPL::init() noexcept(false)
 {
     std::experimental::filesystem::path path(m__modelPath);
     if (std::experimental::filesystem::exists(path)) {
-        std::ifstream file(path);
-        file >> m__model;
+        if( path.extension() == ".cbor"){
+            std::ifstream file(path, std::ios::binary);
+            file >> std::noskipws;
+            
+            file.seekg(0, std::ios::end);
+            const auto file_size = file.tellg();
+            file.seekg(0, std::ios::beg);
+            
+            std::vector<std::uint8_t> cbor_data;
+            cbor_data.reserve(file_size);
+            cbor_data.insert(cbor_data.begin(),std::istream_iterator<char>(file), std::istream_iterator<char>());
+            
+            m__model = nlohmann::json::from_cbor(cbor_data);
```

Comment:
Looks like from_cbor can take istream as input parameter (https://json.nlohmann.me/api/basic_json/from_cbor/). Think cobr_data buffer doesn't need at all.

Reply:
That's neat! I've totally missed it tbh